### PR TITLE
Remove credential info from url in example 

### LINKED ISSUE #3118: HTTP Span Attributes: `http.url` must not contain username / password
## Why

As is stated in the recent [specification change](https://github.com/open-telemetry/opentelemetry-specification/pull/1502) :

> `http.url` MUST NOT contain credentials passed via URL in form of `https://username:password@www.example.com/`. In such case the attribute's value should be `https://www.example.com/`

## What

[This example](https://github.com/open-telemetry/opentelemetry-java/blob/04fbc2e6c52100eff0ada86d6df61adbc8d09569/examples/http/src/main/java/io/opentelemetry/example/http/HttpClient.java#L62 ) could be reviewed (and fixed) to make sure that others do not copy-paste potentially unsafe code.

### PR DESCRIPTION
**Pull request by Hangzhi**
Fix #3118

### CODE REVIEW

File: examples/logging/src/main/java/io/opentelemetry/example/logging/LoggingExporterExample.java
Comment (Hangzhi) [inline examples/logging/src/main/java/io/opentelemetry/example/logging/LoggingExporterExample.java]: Apart from removing the user info in the example above. 
There appears "Cannot find symbol" in this logging example. I did minor modifications base on my understanding.

File: examples/logging/src/main/java/io/opentelemetry/example/logging/LoggingExporterExample.java
Comment (jkwatson) [inline examples/logging/src/main/java/io/opentelemetry/example/logging/LoggingExporterExample.java:22]: Our examples run against the latest release, not against the main branch, so this will get updated after we release 1.1.0. For now, it needs to stay as it was.

Comment (jkwatson) [reply examples/logging/src/main/java/io/opentelemetry/example/logging/LoggingExporterExample.java:22]: In fact, I'm quite concerned that the build seems to be passing with this... something might be amiss with our examples check as a part of CI.

Comment (jkwatson) [reply examples/logging/src/main/java/io/opentelemetry/example/logging/LoggingExporterExample.java:22]: Hm. This is actually annoyingly tricky to get right. We really want our examples to do 2 things:
1) Verify that we haven't broken anything in the stable APIs and
2) Provide good examples that work against the latest official release (not SNAPSHOT builds).

Unfortunately, I don't think it's possible to meet both of these goals for all the examples, since the stable SDK depends on the unstable metrics, so pulling when we build against snapshots for the stable stuff (to verify that we haven't broken anything), we pull in the unstable metrics version as well. 

Note: this is not your problem to solve in this PR...I'm just clarifying for future us. I'll open an issue with these details so we can make this less confusing. :)

Comment (Hangzhi) [reply examples/logging/src/main/java/io/opentelemetry/example/logging/LoggingExporterExample.java:22]: But I think the PR cannot be merged without passing all the tests. What can I do, besides leaving the logging example as what it was and removing the user info in the URL?  It seems that further discussion is needed for this conflict to be resolved.

Comment (jkwatson) [reply examples/logging/src/main/java/io/opentelemetry/example/logging/LoggingExporterExample.java:22]: I'd like to get #3122 fixed before we merge this, so if you're interested in working on that, that would be great.

Comment (jkwatson) [reply examples/logging/src/main/java/io/opentelemetry/example/logging/LoggingExporterExample.java:22]: Note: Now that we've released 1.1.0, this won't be an issue for this PR. Once #3127 has been merged, you can rebase and this PR will be good to go.

Comment (jkwatson) [reply examples/logging/src/main/java/io/opentelemetry/example/logging/LoggingExporterExample.java:22]: (#3122 is still an issue, long term, it's just not a blocker for this PR any more. :)

Comment (Hangzhi) [reply examples/logging/src/main/java/io/opentelemetry/example/logging/LoggingExporterExample.java:22]: Good! I'll do that after #3127 being merged

Comment (jkwatson) [reply examples/logging/src/main/java/io/opentelemetry/example/logging/LoggingExporterExample.java:22]: Merged now! Thanks!

File: examples/http/src/main/java/io/opentelemetry/example/http/HttpClient.java
Comment (kubawach) [inline examples/http/src/main/java/io/opentelemetry/example/http/HttpClient.java:67]: In the spirit of code safety, I'd apply here same logic as suggested for changes in the instrumentation module: 

```
new URI(
    url.getScheme(),
    null,
    url.getHost(),
    url.getPort(),
    url.getPath(),
    url.getQuery(),
    url.getFragment()).toUrl();
```

**Review by kubawach — APPROVED** [review]

File: examples/http/src/main/java/io/opentelemetry/example/http/HttpClient.java
Comment (jkwatson) [reply examples/http/src/main/java/io/opentelemetry/example/http/HttpClient.java:67]: Yes, this is definitely a safer way to do this work. @Hangzhi Can you make this change as well? With this, you can also get rid of the if/else and just do it once using the URI to remove the user info and format correctly.

**Review by jkwatson — APPROVED** [review]
Thanks!

**Pull request merged**
